src/autoschedulers/adams2019/State.h

```diff
@@ -0,0 +1,96 @@
+#ifndef STATE_H
+#define STATE_H
+
+#include "ASLog.h"
+#include "CostModel.h"
+#include "DefaultCostModel.h"
+#include "Featurization.h"
+#include "Halide.h"
+#include "LoopNest.h"
+#include "PerfectHashMap.h"
+#include <map>
+#include <utility>
+
+namespace Halide {
+namespace Internal {
+namespace Autoscheduler {
+
+struct State {
```

Comment:
Suggested top level comment:

A struct representing an intermediate state in the tree search. It represents a partial schedule for some pipeline.

Reply:
Added in 2c918bd